dis_snek/models/discord/channel.py

```diff
@@ -845,6 +846,61 @@ def permissions_for(self, instance: Snowflake_Type) -> Permissions:
 
             return self.permissions_for(instance)
 
+    async def add_permission(
+        self,
+        target: Union["PermissionOverwrite", "models.Role", "models.User"],
+        type: "OverwriteTypes",
+        allow: Optional[List["Permissions"]] = None,
+        deny: Optional[List["Permissions"]] = None,
+        reason: Optional[str] = None,
+    ) -> None:
+        """
+        Add a permission to this channel.
+
+        Args:
+            target: The permission target
+            type: The type of permission overwrite
+            allow: List of permissions to allow
+            deny: List of permissions to deny
+            reason: The reason for this change
+
+        Raises:
+            ValueError: Invalid target for permission
+        """
+        allow = allow or []
+        deny = deny or []
+        if not isinstance(target, PermissionOverwrite):
+            target_type = None
+            if isinstance(target, models.User):
+                target_type = OverwriteTypes.MEMBER
+            elif isinstance(target, models.Role):
+                target_type = OverwriteTypes.ROLE
+            else:
+                raise ValueError("Invalid target for permission")
+            overwrite = PermissionOverwrite(
+                id=target.id, type=target_type, allow=Permissions.NONE, deny=Permissions.NONE
+            )
+            for perm in allow:
+                overwrite.allow |= perm
+            for perm in deny:
+                overwrite.deny |= perm
+        else:
+            overwrite = target
+        if exists := find(lambda x: x.id == overwrite.id and x.type == overwrite.type, self.permission_overwrites):
+            exists.deny = (exists.deny | overwrite.deny) & ~overwrite.allow
+            exists.allow = (exists.allow | overwrite.allow) & ~overwrite.deny
+            return await self.edit_permission(exists, reason)
+
+        permission_overwrites = self.permission_overwrites
+        permission_overwrites.append(overwrite)
+
+        payload = {  # TODO Proper processing
+            "name": self.name,
+            "position": self.position,
+            "permission_overwrites": permission_overwrites,
+        }
+        return await self._edit(payload=payload, reason=reason)
```

Comment:
This part needs to change to align with the latest edit method updates.

Reply:
It now calls `self.edit(permission_overwrites=permission_overwrites)`